README.md

```diff
@@ -25,106 +25,98 @@ Here are some of the benefits of instrumenting applications with Prometheus:
 - Enhanced ability to prevent problems from occurring
 - Increased agility and responsiveness to changes
 
-## Usage
+## Basic Usage
 
 ### Counter
 ```go
-package main
-
-import (
-	"net/http"
-
-	"github.com/go-chi/chi"
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promhttp"
-	"github.com/rabellamy/promstrap"
-)
-
-func main() {
-	go func() {
-		http.Handle("/metrics", promhttp.Handler())
-		http.ListenAndServe(":2112", nil)
-	}()
-
-	candyCounter, _ := promstrap.NewCounterWithLabels(promstrap.MetricsOpts{
-		Namespace: "candy",
-		Name:      "mandm_total",
-		Help:      "Counts the number of M&Ms",
-		Labels:    []string{"color"},
-	})
-
-	prometheus.MustRegister(candyCounter)
-
-	r := chi.NewRouter()
-	r.Get("/m&ms", func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte("M&Ms!!\n"))
-
-		candyCounter.WithLabelValues("red").Add(10)
-		candyCounter.WithLabelValues("orange").Add(5)
-		candyCounter.WithLabelValues("green").Add(5)
-		candyCounter.WithLabelValues("blue").Add(5)
-		candyCounter.WithLabelValues("yellow").Add(10)
-		candyCounter.WithLabelValues("brown").Add(15)
-	})
-
-	http.ListenAndServe(":8080", r)
+// Creates a Prometheus counter with labels to count the number of errors
+errors, err := metrics.NewCounterWithLabels(metrics.CounterOpts{
+	Namespace: "service_name",
+	Name:      "errors_total",
+	Help:      "Number of errors",
+	Labels:    []string{"error"},
+})
+if err != nil {
+	return nil, err
 }
-```
 
-#### Result
-```
-# HELP candy_mandm_total Counts the number of M&Ms
-# TYPE candy_mandm_total counter
-candy_mandm_total{color="blue"} 5
-candy_mandm_total{color="brown"} 15
-candy_mandm_total{color="green"} 5
-candy_mandm_total{color="orange"} 5
-candy_mandm_total{color="red"} 10
-candy_mandm_total{color="yellow"} 10
+
+// Registers errors with the Prometheus DefaultRegisterer
+prometheus.MustRegister(errors)
+
+// Increment by 1
+errors.WithLabelValues("my error").Inc()
 ```
 
 ### Gauge
 ```go
-package main
-
-import (
-	"net/http"
-	"time"
-
-	"github.com/go-chi/chi"
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promhttp"
-	"github.com/rabellamy/promstrap"
-)
-
-func main() {
-	go func() {
-		http.Handle("/metrics", promhttp.Handler())
-		http.ListenAndServe(":2112", nil)
-	}()
-
-	temp, _ := promstrap.NewGaugeWithLabels(promstrap.MetricsOpts{
-		Namespace: "weather",
-		Name:      "current_temperature_celsius",
-		Help:      "The current temperature Celsius",
-		Labels:    []string{"location"},
-	})
-
-	prometheus.MustRegister(temp)
-
-	r := chi.NewRouter()
-	r.Get("/temp", func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte("It's hot!!\n"))
-		temp.WithLabelValues(time.Now().Format("New York, NY")).Set(42)
-	})
-
-	http.ListenAndServe(":8080", r)
+// Creates a Prometheus Gauge with labels to measure the length of a queue
+length, err := metrics.NewGaugeWithLabels(metrics.GaugeOpts{
```

Comment:
@ignatospadov I think `length` should be ok since we are using a **[gauge](https://github.com/rabellamy/promstrap/blob/main/metrics/gauge.go#L18-L21)** in this example with the expectation is that it is dynamic measure.